lib/CL/clGetDeviceInfo.c

```diff
@@ -194,7 +196,29 @@ POname(clGetDeviceInfo)(cl_device_id   device,
     POCL_RETURN_GETINFO(cl_bool, device->compiler_available);
   case CL_DEVICE_EXECUTION_CAPABILITIES            :
     POCL_RETURN_GETINFO(cl_device_exec_capabilities, device->execution_capabilities);
-   
+
+  case CL_DEVICE_COMPUTE_CAPABILITY_MAJOR_NV       :
+  case CL_DEVICE_COMPUTE_CAPABILITY_MINOR_NV       :
+  case CL_DEVICE_REGISTERS_PER_BLOCK_NV            :
+  case CL_DEVICE_WARP_SIZE_NV                      :
+  case CL_DEVICE_ATTRIBUTE_ASYNC_ENGINE_COUNT_NV   :
+  case CL_DEVICE_PCI_BUS_ID_NV                     :
+  case CL_DEVICE_PCI_SLOT_ID_NV                    :
+  case CL_DEVICE_PCI_DOMAIN_ID_NV                  :
+    {
+      int res;
+      res = pocl_cuda_handle_cl_nv_device_attribute_query(device, param_name);
```

Comment:
This calls CUDA driver functions unconditionally. Doesn't this break if the CUDA driver is not enabled&built? Did you test without CUDA?

Reply:
I tested with and without CUDA, and it seems to have worked like I would expect. I have made the extension conditional on `BUILD_CUDA` to make it a bit cleaner though.

Reply:
`clinfo -a/-A` appear to work fine with and without CUDA.